### PR TITLE
[FIX] No newline was printed with the exit string when Ctrl+D

### PR DESCRIPTION
**Pull request by itislu**
- [x] Still need to solve #188 
- [x] First merge #192 to then test with stricter memory leak tests.

### CODE REVIEW

**Comment by itislu** [comment]
This is a good example that there is something wrong.
It's a tiny change and not related to the failed test case at all, yet it fails.
If we run the tester a few times, at some point it will probably pass.
But that's not a good sign.

**Comment by LeaYeh** [comment]
Can not seperate exit msg and newline in this way, in subshell Ctrl+D would not have newline

**Comment by itislu** [comment]
I don't understand yet why there are 2 newlines printed when first doing Ctrl+C and then Ctrl+D...

**Comment by itislu** [comment]
Make sure to have readline library v8.1 installed, as on the school computers.
I had v8.0 installed and some newline behaviors were different.

You can check by putting the following at the beginning of the main function:
```c
printf("%s\n", rl_library_version);
```

**Comment by itislu** [comment]
If we want, we can implement signal messages for every existing signal the same way as bash does it: https://github.com/bminor/bash/blob/master/siglist.c

Atm, we only print a message for `SIGQUIT` and `SIGSEGV`.

File: source/backend/executor/handle_pipeline.c [hunk 1]
```diff
@@ -67,7 +56,7 @@ void	exec_pipeline(t_shell *shell, t_list_d **cmd_table_node)
 		else
 		{
 			if (need_pipe(*cmd_table_node) && !create_pipe(&shell->new_pipe))
-				raise_error_to_all_subprocess(shell, 129, ERROR_CREATE_PIPE);
+				raise_error_to_all_subprocess(shell, 129, ERROR_CREATE_PIPE);	// Magic number
```
Comment (itislu) [inline source/backend/executor/handle_pipeline.c:59]: @LeaYeh What does 129 stand for?

Comment (itislu) [reply source/backend/executor/handle_pipeline.c:62]: We found out it stands for `SIGHUP`.
Bash's exit code is also `129` if it fails to create more pipes.

**Pull request merged**